liquibase-core/src/main/java/liquibase/sqlgenerator/core/CreateProcedureGenerator.java

```diff
@@ -70,7 +70,7 @@ public ValidationErrors validate(CreateProcedureStatement statement, Database da
 
         procedureText = removeTrailingDelimiter(procedureText, statement.getEndDelimiter());
 
-        if (database instanceof MSSQLDatabase && procedureText.toLowerCase().contains("merge") && !procedureText.endsWith(";")) { //mssql "AS MERGE" procedures need a trailing ; (regardless of the end delimiter)
+        if (database instanceof MSSQLDatabase && procedureText != null && procedureText.toLowerCase().contains("merge") && !procedureText.endsWith(";")) { //mssql "AS MERGE" procedures need a trailing ; (regardless of the end delimiter)
```

Comment:
Wrapping procedureText.toLowerCase() in StringUtils.trimToEmpty(procedureText).toLowerCase() is maybe more understandable?